src/app.test.ts

```diff
@@ -1,51 +1,267 @@
 
 import app from './app'
 import mapboxRasterTilejson from './fixtures/good-tilejson/mapbox_raster_tilejson.json'
-import { getTilesetId } from './lib/tilestore'
+import { getTilesetId } from './lib/utils'
+import { TileJSON, validateTileJSON } from './lib/tilejson'
+import { server as mockTileServer } from './mocks/server'
+
+import { IdResource } from './api'
 
 tmp.setGracefulCleanup()
 
-beforeEach((done, t) => {
+type TestContext = {
+  accessDb: (cb: (db: DatabaseInstance) => void) => void
+  server: FastifyInstance
+  sampleTileJSON: TileJSON
+}
+
+function assertSampleTileJSONIsValid(data: unknown): asserts data is TileJSON {
+  if (!validateTileJSON(data)) {
+    const message = `Sample input does not conform to TileJSON schema spec: ${JSON.stringify(
+      validateTileJSON.errors,
+      null,
+      2
+    )}`
+
+    throw new Error(message)
+  }
+}
+
+before(() => {
+  // Check if prisma/migrations directory exists in project
+  if (!fs.existsSync(path.resolve(__dirname, '../prisma/migrations'))) {
+    throw new Error(
+      'Could not find prisma migrations directory. Make sure you run `npm run prisma:migrate-dev -- --name MIGRATION_NAME_HERE` first!'
+    )
+  }
+
+  assertSampleTileJSONIsValid(mapboxRasterTilejson)
+
+  mockTileServer.listen()
+})
+
+beforeEach((t) => {
   const { name: dataDir } = tmp.dirSync({ unsafeCleanup: true })
-  t.context.app = app({ logger: false }, { dataDir })
-  done()
+
+  const dbPath = path.resolve(dataDir, 'test.db')
+
+  t.context = {
+    accessDb: (cb: (db: DatabaseInstance) => void) => {
+      const db = new DB(dbPath, { readonly: true })
+      cb(db)
+      db.close()
+    },
+    server: app({ logger: false }, { dbPath }),
+    sampleTileJSON: mapboxRasterTilejson,
+  }
+})
+
+afterEach((t) => {
+  t.context.server.close()
+  mockTileServer.resetHandlers()
 })
 
-afterEach((done, t) => {
-  t.context.app.close().then(done)
+teardown(() => {
+  mockTileServer.close()
 })
 
 test('GET /tilesets (empty)', async (t) => {
-  const { app } = t.context
+  const { server } = t.context as TestContext
+
+  const response = await server.inject({ method: 'GET', url: '/tilesets' })
+
+  t.equal(response.statusCode, 200, 'returns a status code of 200')
 
-  const response = await app.inject({ method: 'GET', url: '/tilesets' })
-  t.strictEqual(response.statusCode, 200, 'returns a status code of 200')
-  t.strictEqual(
+  t.equal(
     response.headers['content-type'],
     'application/json; charset=utf-8',
     'returns correct content-type header'
   )
-  t.deepEqual(response.json(), [], 'returns empty array')
+
+  t.same(response.json(), [], 'returns empty array')
+
+  t.end()
+})
+
+test('GET /tilesets (not empty)', async (t) => {
+  const { sampleTileJSON, server } = t.context as TestContext
+
+  await server.inject({
+    method: 'POST',
+    url: '/tilesets',
+    payload: sampleTileJSON,
+  })
+
+  const expectedId = getTilesetId(sampleTileJSON)
```

Comment:
I think we should hardcode this value (rather than calculate it with this function), to ensure that our getTilesetId function is working as expected.